pkg/api/handlers/compat/auth.go

```diff
@@ -50,9 +50,19 @@ func Auth(w http.ResponseWriter, r *http.Request) {
 			Status:        "Login Succeeded",
 		})
 	} else {
-		utils.WriteResponse(w, http.StatusBadRequest, entities.AuthReport{
-			IdentityToken: "",
-			Status:        "login attempt to " + authConfig.ServerAddress + " failed with status: " + err.Error(),
+		var msg string
+
+		var unauthErr DockerClient.ErrUnauthorizedForCredentials
+		if errors.As(err, &unauthErr) {
+			msg = "401 Unauthorized"
+		} else {
+			msg = err.Error()
+		}
+
+		utils.WriteResponse(w, http.StatusInternalServerError, struct {
```

Comment:
This should be a 401 and not 500 code when you match ErrUnauthorizedForCredentials

Reply:
Otherwise you return 500 but it says 401 in the error message

Reply:
@Luap99 totally agree but `moby` returns `500` so just to be sure I return the same.

Reply:
Alright, bug for bug compatibility then

Reply:
also returning `401` would make this code work:
```go
	_, err = cli.RegistryLogin(ctx, types.AuthConfig{Username: username, Password: password, ServerAddress: registry})
	if dockerClient.IsErrUnauthorized(err) {
		return ErrUnauthorized
	}
```
right now I have to do this awful awful thing:
```go
	_, err = cli.RegistryLogin(ctx, types.AuthConfig{Username: username, Password: password, ServerAddress: registry})
	if err != nil && strings.Contains(err.Error(), "401 Unauthorized") {
		return ErrUnauthorized
	}
```
Still I am afraid doing something differently than `moby`.

Reply:
I'll add comment to src code.